napari_arboretum/plugin.py

```diff
@@ -84,41 +87,71 @@ def _append_mouse_callback(self, layer: napari.layers.Tracks):
 
         @layer.mouse_drag_callbacks.append
         def show_tree(layer, event):
+            self.layer = layer
 
             # cursor_position = layer.position
             cursor_position = self._viewer.cursor.position
 
             # fix to return the track ID using the world coordinates returned
             # by `viewer.cursor.position`
-            track_id = layer.get_value(cursor_position, world=True)
-            root, subgraph_nodes = build_subgraph(layer, track_id)
+            self.track_id = layer.get_value(cursor_position, world=True)
+
+            root, subgraph_nodes = build_subgraph(layer, self.track_id)
 
             if not subgraph_nodes:
-                print(track_id, root, subgraph_nodes)
+                print(self.track_id, root, subgraph_nodes)
                 return
 
-            edges, annotations = layout_subgraph(root, subgraph_nodes)
-            self.draw_graph(track_id, edges, annotations)
-
-    def draw_graph(self, track_id, edges, annotations):
-        """Plot graph on the plugin canvas."""
+            self.edges, self.annotations = layout_subgraph(root, subgraph_nodes)
+            self.draw_graph()
+
+    def update_colors(self):
+        """
+        Update colors on a list of edges from the colors they have in a
+        track layer. Note that this updates the colors in-place on ``edges``.
+        """
+        for e in self.edges:
+            if e.id is not None:
+                color = self.layer.track_colors[
+                    np.where(self.layer.properties["track_id"] == e.id)
+                ][-1]
```

Comment:
would `self.layer.track_colors[self.layer.properties["track_id"] == e.id][-1]` achieve the same thing?